### PR TITLE
Add support for dynamic renderer addition and swap

### PR DESCRIPTION
**Pull request by nmellado**
PR is ready.
Adds: 
 * GUI to select active renderer in Gui::MainApp
 * Fallback in Gui::Viewer: when no renderer is set, use ForwardRenderer as default
 * Update PluginInterface with a new handle to create renderers from plugins.

### CODE REVIEW

**Comment by nmellado** [comment]
PR ready for review, please have a look.

**Comment by nmellado** [comment]
Fixes #63

**Comment by nmellado** [comment]
Fixes #219

File: src/PluginBase/RadiumPluginInterface.hpp [hunk 1]
```diff
@@ -106,6 +113,17 @@ namespace Ra
...
+            virtual bool doAddRenderer() { return false; }
+
+            /**
+             * @brief addRenderers
+             *
+             * \param rds the TODO
+             * \warning Allocated renderers are given to the application and
+             * SHOULD not be destroyed by the plugin
+             */
+            virtual void addRenderers(std::vector<Engine::Renderer*> */*rds*/) {}
```
Comment (dlyr) [inline src/PluginBase/RadiumPluginInterface.hpp]: TODO : Use shared_ptr to share the ownership of renderers between plugin and viewer

File: Applications/MainApplication/Gui/MainWindow.cpp [hunk 2]
```diff
@@ -576,6 +568,12 @@ namespace Ra
                                              QItemSelectionModel::ClearAndSelect | QItemSelectionModel::Current);
     }
 
+    void Gui::MainWindow::onGLInitialized()
+    {
+        // set renderers once OpenGL is configured
+        addRenderer("Forward Renderer", new Engine::ForwardRenderer());
```
Comment (dlyr) [inline Applications/MainApplication/Gui/MainWindow.cpp]: TODO : Use shared_ptr to share the ownership of renderers between mainapp and viewer

File: Applications/MainApplication/MainApplication.cpp [hunk 3]
```diff
@@ -497,6 +497,18 @@ namespace Ra
                         ++pluginCpt;
                         loadedPlugin->registerPlugin( context );
                         m_mainWindow->updateUi( loadedPlugin );
+
+                        if(loadedPlugin->doAddRenderer())
+                        {
```
Comment (dlyr) [inline Applications/MainApplication/MainApplication.cpp]: change this according to the shared_ptr alternative

**Comment by MathiasPaulin** [comment]
Mis à part les warning ci-dessous (à corriger, ça fait plus prore :p), tout va bien pour moi ....
Je validerais la PR quand ces warnings n'y seront plus ...

Plugins/Animation/src/AnimationComponent.cpp:187:36: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
src/GuiBase/SelectionManager/SelectionManager.hpp:43:18: warning: 'Ra::GuiBase::SelectionManager::select' hides overloaded virtual functions [-Woverloaded-virtual]
src/AnimationComponent.hpp:121:13: warning: private field 'm_selectedBone' is not used [-Wunused-private-field]
src/Engine/Renderer/Renderers/ExperimentalRenderer.cpp:35:26: warning: unused variable 'buffers' [-Wunused-const-variable]
src/Engine/Renderer/Texture/Texture.hpp:207:18: warning: private field 'm_textureId' is not used [-Wunused-private-field]

**Comment by nmellado** [comment]
@MathiasPaulin I"ve fixed the warnings.

**Pull request merged**